WWTMVC5/WWTWeb/demmars.aspx

```diff
@@ -1,47 +1,46 @@
-<%@ Page Language="C#" ContentType="text/plain" %>
+<%@ Page Language="C#" ContentType="application/octet-stream" CodeFile="DemMars.aspx.cs" Inherits="WWTMVC5.WWTWeb.DemMars" %>
+<%@ Import Namespace="System.Drawing" %>
+<%@ Import Namespace="System.Drawing.Text" %>
+<%@ Import Namespace="System.Drawing.Imaging" %>
+<%@ Import Namespace="System.Drawing.Drawing2D" %>
+<%@ Import Namespace="System.Collections.Generic" %>
 <%@ Import Namespace="System.IO" %>
-<%@ Import Namespace="WWTMVC5.WWTWeb" %>
-<%@ Import Namespace="WWTWebservices" %>
+<%@ Import Namespace="System.IO" %>
+<%@ Import Namespace="System.Security.Cryptography" %>  
+<%@ Import Namespace="System.Diagnostics" %>
+<%@ Import Namespace="WWTWebservices" %>d
 <%
+
     string query = Request.Params["Q"];
     string[] values = query.Split(',');   
     int level = Convert.ToInt32(values[0]);
     int tileX = Convert.ToInt32(values[1]);
     int tileY = Convert.ToInt32(values[2]);
-	
-    //string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
-    string DSSTileCache = WWTUtil.GetCurrentConfigShare("DSSTileCache", true);
-
-    DSSTileCache = @"\\wwt-sql01\dsstilecache";
-
-    string filename = String.Format(DSSTileCache  + "\\wwtcache\\mars\\dem\\{0}\\{2}\\{1}_{2}.dem", level, tileX, tileY);
-
-    //string path = String.Format(DSSTileCache + "\\wwtcache\\mars\\dem\\{0}\\{2}", level, tileX, tileY);
 
-
-
-    if (!File.Exists(filename))
+    if (level < 18)
     {
-       // try
-       // {
-       //     if (!Directory.Exists(filename))
-       //     {
-       //         Directory.CreateDirectory(path);
-       //     }
-            
-       //     WebClient webclient = new WebClient();
-
-       //     string url = string.Format("http://wwt.nasa.gov/wwt/p/mars_toast_dem_32f/{0}/{1}/{2}.toast_dem_v1",  level, tileX, tileY);
-
-       //     webclient.DownloadFile(url, filename);
-      //  }
-      //  catch
-       // {
-            Response.StatusCode = 404;
+    //    Response.ContentType = "image/png";
+
+        UInt32 index = ComputeHash(level, tileX, tileY) % 400;
+
+        Stream s = PlateFile2.GetFileStream(String.Format(@"\\wwtfiles.file.core.windows.net\wwtmars\MarsDem\marsToastDem.plate", index), -1, level, tileX, tileY);
```

Comment:
`index` isn't used in the `Format` here?

Reply:
Good catch. Appears to be so. Once we get the code moved out of the aspx files, then things like that can be caught very easily by some new tooling built into .NET.